### PR TITLE
Add requestTypes tag support

### PR DESCRIPTION
**Pull request by Prateeknandle**
Signed-off-by: Prateeknandle <prateeknandle@gmail.com>

### CODE REVIEW

File: proposals/add_requestTypes_tag_support [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+- [Kyverno Design Proposal - Adding Request Types to match and exclude](#kyverno-design-proposal---adding-request-types-to-match-and-exclude)
+  * [Introduction](#introduction)
+  * [Proposal](#proposal)
+  * [Example](#example)
+  * [Implementation](#implementation)
+
+## Introduction
+
+Until now we use `request.operation` under `condition.key`, to filter a request, after this proposal we'll use a tag `requestTypes` in `match` or `exclude` block to match/exclude specific requests coming from admission review requests.
```
Comment (chipzoller) [inline proposals/add_requestTypes_tag_support:17]: This would be `preconditions[]` as it could be in key or value.

**Review by chipzoller — COMMENTED** [review]
What will the default behavior be if `requestTypes` is not present? This change must allow backwards compatibility with existing policies which do not specify `requestTypes` and use preconditions.

**Comment by Prateeknandle** [comment]
> What will the default behavior be if `requestTypes` is not present?

If `requestTypes` is not defined, Policy will behave same as now behaves when we don't filter out the request using `request.operation`, @realshuting is it correct statement?

> This change must allow backwards compatibility with existing policies which do not specify `requestTypes` and use preconditions.

`requestTypes` is a separate field, it'll not come under `any/all`, [code](https://github.com/kyverno/kyverno/pull/3619/files#diff-4161872653e054be4b55c2477d73cce4f0333b5123d0840b7f13ea6d5879df50R17).

File: proposals/add_requestTypes_tag_support (hunk 1 above)
Comment (Prateeknandle) [reply proposals/add_requestTypes_tag_support]: updated

**Comment by chipzoller** [comment]
See https://github.com/kyverno/kyverno/issues/3110 for an example of where even this behavior internally isn't consistent depending on the contents of the rule. `requestTypes` should improve upon such behavior.

> `requestTypes` is a separate field, it'll not come under `any`/`all`, code.

Yes, that's fine, but my point was this addition must preserve operation of current policies if it is absent.

**Comment by chipzoller** [comment]
This has been implemented in https://github.com/kyverno/kyverno/pull/6658

**Pull request closed without merging**